controllers/fyController.js

```diff
@@ -0,0 +1,309 @@
+      console.log("SORT ERROR\n");
+      return res.send({ status: STATUS_CODE.UNKNOWN_SORT });
+    }
+  }
+
+  // create where string
+  var where = '';
+  const where_list = [name_search, email_search, visions_filter, vuceptor_search, vuceptor_filter];
+  const prefix_list = ['students.name = ', 'students.email = ', 'students.visions = ', 'users.name = ', 'users.name = '];
+  for (let m = 0; m < where_list.length; m++){
+    cond = where_list[m];
+    prefix = prefix_list[m];
+
+    if(cond !== ''){
+      let tmp = '';
+
+      if (where !== ''){
+        // initialize
+        if (m == (where_list.length - 1)){ // visions_filter
+          tmp = prefix + cond[0];
+        } else{
+          tmp = prefix + '\'' + cond[0] + '\'';
+        }
+
+        if (m == (where_list.length - 1)){ // visions_filter
+          for(let k = 1; k < cond.length; k++){
+            tmp = tmp + ' OR ' + prefix + cond[k];
+          }
+        } else {
+          for(let k = 1; k < cond.length; k++){
+            tmp = tmp + ' OR ' + prefix + '\'' + cond[k] + '\'';
+          }
+        }
+
+        where = where + ' AND (' + tmp + ')';
+      } else {
+        let tmp = '';
+
+        // initialize
+        if (m == (where_list.length - 1)){ // visions_filter
+          tmp = prefix + cond[0];
+        } else{
+          tmp = prefix + '\'' + cond[0] + '\'';
+        }
+
+        if (m == (where_list.length - 1)){ // visions_filter
+          for(let k = 1; k < cond.length; k++){
+            tmp = tmp + ' OR ' + prefix + cond[k];
+          }
+        } else{
+          for(let k = 1; k < cond.length; k++){
+            tmp = tmp + ' OR ' + prefix + '\'' + cond[k] + '\'';
+          }
+        }
+
+        where = ' WHERE (' + tmp + ')';
+      }
+    }
+  }
+
+  // ORDER BY A, B will first order database by A, if A is the same then order by B
+  // if condition_order is not passed in, it will order with priorty of name -> email -> visions
+  // if condition_order is given, it will only contain sorting conditions within condition_order 
+  var orderby = '';
+  var orderby_list = []
+  if (condition_order == null){
+    orderby_list = [name_sort, email_sort, visions_sort];
+  } else {
+    for(let i = 0; i < condition_order.length; i++){
+      let cond = condition_order[i];
+  
+      switch (cond) {
+        case "name_sort":
+          orderby_list.push(name_sort);
+          break;
+        case "email_sort":
+          orderby_list.push(email_sort);
+          break;
+        case "visions_sort":
+          orderby_list.push(visions_sort);
+      }
+    }
+  }
+
+  // create ORDER BY string
+  orderby_list.forEach(order => {
+    if(order !== ''){
+      if (orderby !== ''){
+        orderby = orderby + ' ,' + order;
+      } else {
+        orderby = ' ORDER BY' + order;
+      }
+    }
+  })
+
+  //calculate the number of pages
+  const queryCount = "SELECT COUNT(*) AS count FROM students";
+  var pages = 0;
```

Comment:
pages is not updated when the user changes conditions.
<img width="727" alt="image" src="https://user-images.githubusercontent.com/107736505/197244385-81b0c69a-5fe8-4801-8bef-d00f80c8df49.png">

Reply:
great point!
Fixed the bug in userController and fyController